sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/ApplicationInsightsRestClient.cs

```diff
@@ -16,18 +18,28 @@ internal partial class ApplicationInsightsRestClient
         /// This operation sends a sequence of telemetry events that will be monitored by Azure Monitor.
         /// </summary>
         /// <param name="body">The list of telemetry events to track.</param>
+        /// <param name="storage">Storage object</param>
         /// <param name="cancellationToken">The cancellation token to use.</param>
         /// <returns></returns>
-        internal async Task<int> InternalTrackAsync(IEnumerable<TelemetryItem> body, CancellationToken cancellationToken = default)
+        internal async Task<int> InternalTrackAsync(IEnumerable<TelemetryItem> body, IPersistentStorage storage, CancellationToken cancellationToken = default)
         {
             if (body == null)
             {
                 throw new ArgumentNullException(nameof(body));
             }
 
             using var message = CreateTrackRequest(body);
-            message.SetProperty("TelemetryItems", body);
-            await _pipeline.SendAsync(message, cancellationToken).ConfigureAwait(false);
+            message.SetProperty("TelemetryItemCount", body.Count());
+
+            try
+            {
+                await _pipeline.SendAsync(message, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                AzureMonitorExporterEventSource.Log.Write($"FailedToSend{EventLevelSuffix.Error}", ex.LogAsyncException());
+                HttpPipelineHelper.SaveTelemetryToStorage(storage, message);
```

Comment:
Is there a way to know if the exception indicates an retriable error (in this case we'll store the data) or non-retriable error (in this case we'll drop the data)?